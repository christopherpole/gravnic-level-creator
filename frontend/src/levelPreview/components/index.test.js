import React from 'react';
import { configure, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import { ENTITIES } from 'gravnic-game';

import { LevelPreview } from './index';

configure({ adapter: new Adapter() });

describe('<LevelPreview />', () => {
  let props;

  beforeEach(() => {
    props = {
      entitiesData: {
        1: {
          xPos: 10,
          yPos: 10,
          id: 123,
          color: '#ff0000',
          entityId: ENTITIES.BLOCK,
          isMovableEntity: true,
        },
      },
      makeMoveAction: () => {},
      setInitialStateAction: () => {},
      entitiesMoving: false,
      levelComplete: false,
    };
  });

  it('Renders without exploding', () => {
    const gameArea = shallow(<LevelPreview {...props} />);

    expect(gameArea).toHaveLength(1);
  });

  it("Matches the current snapshot if the level isn't complete", () => {
    const gameArea = shallow(<LevelPreview {...props} />);

    expect(toJson(gameArea)).toMatchSnapshot();
  });

  it('Matches the current snapshot if the level is complete', () => {
    const gameArea = shallow(<LevelPreview {...props} levelComplete />);

    expect(toJson(gameArea)).toMatchSnapshot();
  });

  it('Calls the make move action when the user presses the arrow keys on the keyboard');
});
