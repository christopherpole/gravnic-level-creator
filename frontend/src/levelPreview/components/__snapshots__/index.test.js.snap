// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LevelPreview /> Matches the current snapshot if the level is complete 1`] = `
<styled.div
  id="level-preview"
>
  <styled.div>
    <Connect(Component)
      entityId="BLOCK"
      isMovableEntity={true}
      key="1"
      xPos={10}
      yPos={10}
    />
  </styled.div>
  <styled.div
    id="level-complete-overlay"
  >
    <p>
      Level Complete!
    </p>
  </styled.div>
</styled.div>
`;

exports[`<LevelPreview /> Matches the current snapshot if the level isn't complete 1`] = `
<styled.div
  id="level-preview"
>
  <styled.div>
    <Connect(Component)
      entityId="BLOCK"
      isMovableEntity={true}
      key="1"
      xPos={10}
      yPos={10}
    />
  </styled.div>
</styled.div>
`;
